packages/teams-js/src/internal/appIdValidation.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * This function can be used to validate if a string is a "valid" app id.
+ * Valid is a relative term, in this case. Truly valid app ids are UUIDs as documented in the schema:
+ * https://learn.microsoft.com/en-us/microsoftteams/platform/resources/schema/manifest-schema#id
+ * However, there are some older internal/hard-coded apps which violate this schema and use names like com.microsoft.teamspace.tab.youtube.
+ * For compatibility with these legacy apps, we unfortunately cannot securely and completely validate app ids as UUIDs. Based
+ * on this, the validation is limited to checking for script tags, length, and non-printable characters.
+ *
+ * @param potentialAppId A string to check if it's a "valid" app id
+ * @throws Error with a message describing the exact validation violation
+ */
+export function validateStringAsAppId(potentialAppId: string): void {
+  if (doesStringContainScriptTags(potentialAppId)) {
+    throw new Error(`Potential app id (${potentialAppId}) is invalid; it contains script tags.`);
+  }
+
+  if (!isStringWithinAppIdLengthLimits(potentialAppId)) {
+    throw new Error(
+      `Potential app id (${potentialAppId}) is invalid; its length ${potentialAppId.length} is not within the length limits (${minimumValidAppIdLength}-${maximumValidAppIdLength}).`,
+    );
+  }
+
+  if (doesStringContainNonPrintableCharacters(potentialAppId)) {
+    throw new Error(`Potential app id (${potentialAppId}) is invalid; it contains non-printable characters.`);
+  }
+}
+
+export function doesStringContainScriptTags(str: string): boolean {
```

Comment:
I feel like we've also had to write code combing URLs for script tags, might be an opportunity to share

Reply:
I was unaware of this but I have now found that code. In the process I have also found code that already does app id validation. However, it is not as thorough, does not mention app IDs at all (it just says "id"), and doesn't give specific error messages on failures, just a generic error message.

My plan is to go forward with this code and then make a subsequent change to unify the app validation code using the one that has the most thorough validation and clearest error messages.